src/engine.ts

```diff
@@ -55,10 +60,18 @@ class Engine {
     };
 
     try {
-      browserPage = await this.createPage(options.emulateOptions, options.interceptors);
+      browserPage = await this.createPage(options.emulateOptions, options.interceptors, options.followRedirects);
       const navigationResult = await browserPage.goto(options.url, {waitUntil: options.waitMethod});
 
-      if (navigationResult) {
+      if (options.followRedirects && browserPage.redirect) {
+        const redirectRequest = browserPage.redirect;
+        const headers = redirectRequest.headers();
+        const status = redirectRequest.status();
+        renderStatus.status = status;
+        renderStatus.html = `Moved ${headers.location}`;
```

Comment:
Don't make this hardcoded like this, most of the time body content for is 3xx useless. But they still should be able to provide whatever they want. Please use the real HTML body you received from the real application.